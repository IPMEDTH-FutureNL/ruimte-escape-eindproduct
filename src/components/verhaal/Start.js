import React, { useState } from 'react';
import { useNavigate } from "react-router-dom";
import '../../css/start.css';
import Note from "../../img/gameplayImages/snippets/note.png"
import Astronaut from "../../img/gameplayImages/characters/character2_standard_schuin_mouthopen.svg"
import { FaInfoCircle } from "react-icons/fa";
import { FaAngleRight } from "react-icons/fa";
import TextCloud from '../../img/gameplayImages/snippets/textcloud.png'
// import IntroSound from '../../sound/Intro.mp3';


const Start = () => {
    // const audio = new Audio(IntroSound);
    const [text, setText] = useState("");
    const [playerState, setPlayerState] = useState("false");
    const [count, setCount] = useState(0);
    const navigate = useNavigate();
    
    const showPlayerForm = (number) => {
        let formOne = document.getElementById("container__eenSpeler");
        let formTwo = document.getElementById("container__tweeSpelers");
        if(number === 1){
            setPlayerState("false");
            formOne.style.display = "block";
            formTwo.style.display = "none";
        }else{
            setPlayerState("true");
            formOne.style.display = "none";
            formTwo.style.display = "block";
        }

    }
    
    const players = (e) => {
        e.preventDefault();
        // audio.play();
        let players = getPlayerNames();
        let introduction = document.getElementById("container__introduction");
        let gameplay = document.getElementById("container__gameplay");


        let playerOne = "";
        let playerTwo = "";

        playerOne = players[0];
        playerTwo = players[1];
        localStorage.setItem("playerOne", playerOne);
        localStorage.setItem("playerTwo", playerTwo);

        introduction.style.display = "none";
        gameplay.style.display = "block";
        localStorage.setItem('punten', 30);
  
        if(localStorage.getItem("playerTwo") === "undefined"){
            localStorage.setItem("playerTwo", "");
            setText("Welkom " + localStorage.getItem("playerOne") + " op het ruimteschip! Ik ben Max een van de astronauten op dit schip.")
        }else{
            setText("Welkom " + localStorage.getItem("playerOne") + " en " + localStorage.getItem("playerTwo") + " op het ruimteschip! Ik ben Max een van de astronauten op dit schip.")
        }
       
    }

    const getPlayerNames = () => {

        let input = "";

        if(playerState === "false"){
            input = document.getElementsByClassName("player__form-name-speler");
        }else{
            input = document.getElementsByClassName("player__form-name");
        }
        let players = [];
        let currentPlayer = ""
       

        for(let i = 0; i < input.length; i++){
            currentPlayer = input[i].value;
            players.push(currentPlayer);
        }

        return players;
    }

    const story = (newText) => {
        setCount(count + newText);

        switch(count){
            case 0:
                setText("We zijn bijna klaar om koers te zetten naar Mars!")
                break;
            case 1:
                if(localStorage.getItem("playerTwo") != null){
                    setText("We hebben alleen nog een probleem, de kaart klopt niet. Ik heb je hulp nodig!");
                }else{
                    setText("We hebben alleen nog een probleem, de kaart klopt niet. Ik heb jullie hulp nodig!");
                }
                break;
            case 2:
                navigate('/introPuzzle');
                break;
        }
    }

    return(
        <section className="container">
            <section className="containter__start">
                <section className="container__introduction" id="container__introduction">
                    <img src={Note} alt="note with rules" className="escaperoom__rules-note" draggable="false"></img>
                    <div className="escaperoom__rules">
                        <h2>Welkom bij de missie naar Mars!</h2>
                        <p>Voordat we beginnen nog een paar korte tips!</p>
                        <div className="rule">
                            <button className='rule-hint'> Tips</button>
                            <p className="hint-text">De tip knop, hier kan je tips krijgen over de puzzels de kosten alleen 10 punten!</p>
                        </div>
                        <div className="rule">
                            <div className="rule-clickable" />
                            De groene rondjes in de escaperoom, betekenen dat je erop kan klikken!
                        </div>
                        <div className="rule">
                            Met hoeveel personen zijn jullie?
                        </div>
                        <div className="button__group">
                            <button className="button primary" onClick={() => showPlayerForm(1)}>1</button>
                            <button className="button primary" onClick={() => showPlayerForm(2)}>2</button>
                        </div>
                    </div>


                    <div className="container__eenSpeler" id="container__eenSpeler">
                        <form className="player__form first" onSubmit={(e) => players(e)}>
                            <div className="inputdiv">
                                <label className="player__form-label">Naam</label>
                                <input className="player__form-name-speler"></input>
                            </div>
                            <input type="submit" value="Verder" className='button primary submit'/>
                        </form>
                    </div>

                    <div className="container__tweeSpelers" id="container__tweeSpelers">
                        <form className="player__form second" onSubmit={(e) => players(e)}>
                            <div className="inputdiv">
                                <label className="player__form-label">Naam</label>
                                <input className="player__form-name"></input>
                            </div>
                            <div className="inputdiv">
                                <label className="player__form-label">Naam</label>
                                <input className="player__form-name"></input>
                            </div>
                            <div className="inputdiv">
                                <input type="submit" value="Verder" className='button primary submitSecond'/>
                            </div>
                        </form>
                    </div>
                </section>

                <section className="container__gameplay" id="container__gameplay">
                    <div className="container__astronaut">
                        <div className="astronaut">
                            <img src={Astronaut} alt="astronaut" className="astronaut__image" draggable="false"></img>
                        </div>
                        <div className="textcloud">
                            <img src={TextCloud} alt="tekst balloon" className="textcloud__image" draggable="false"></img>
                            <div className="textcloud__field">
                                <p className="textcloud__text">{text}</p>
                                <button className="continue" onClick={() => story(1)}>Verder</button>
                            </div>
                        </div>
                    </div>
                </section>
            </section>
        </section>
    )
}

export default Start;