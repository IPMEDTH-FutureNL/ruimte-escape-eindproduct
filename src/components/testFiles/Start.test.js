import renderer from 'react-test-renderer';
import Start from "../verhaal/Start";
import {BrowserRouter as Router} from 'react-router-dom';

it('Test if start component is rendered', () => {
    const component = renderer.create(
        <Router>
            <Start></Start>,
        </Router>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})