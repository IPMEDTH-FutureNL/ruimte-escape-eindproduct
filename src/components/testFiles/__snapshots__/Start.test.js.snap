// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if start component is rendered 1`] = `
Array [
  <section
    className="container"
  >
    <section
      className="containter__start"
    >
      <section
        className="container__introduction"
        id="container__introduction"
      >
        <img
          alt="note with rules"
          className="escaperoom__rules-note"
          src="note.png"
        />
        <div
          className="escaperoom__rules"
        >
          <h2>
            Welkom bij de ruimte-escape
          </h2>
          <p>
            Voordat we beginnen nog een paar korte tips!
          </p>
          <div
            className="rule"
          >
            <svg
              className="rule-hint"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
              />
            </svg>
            De hint icoon, hier kan je hints krijgen over de puzzels!
          </div>
          <div
            className="rule"
          >
            <svg
              className="rule-text"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 256 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              />
            </svg>
            Wanneer de pijl in een tekst balloon is geplaatst kan je hiermee verder in het verhaal!
          </div>
          <div
            className="rule"
          >
            <div
              className="rule-clickable"
            />
            De groene rondjes in binnen de escaperoom, betekenen dat je erop kan klikken!
          </div>
        </div>
        <form
          className="player__form"
          onSubmit={[Function]}
        >
          <div
            className="inputdiv"
          >
            <label
              className="player__form-label"
            >
              Speler 1
            </label>
            <input
              className="player__form-name"
            />
          </div>
          <div
            className="inputdiv"
          >
            <label
              className="player__form-label"
            >
              Speler 2
            </label>
            <input
              className="player__form-name"
            />
          </div>
          <div
            className="inputdiv"
          >
            <input
              className="player__form-submit"
              type="submit"
              value="Verder"
            />
          </div>
        </form>
      </section>
      <section
        className="container__gameplay"
        id="container__gameplay"
      >
        <div
          className="container__astronaut"
        >
          <div
            className="astronaut"
          >
            <img
              alt="astronaut"
              className="astronaut__image"
              src="character2_standard_schuin_mouthopen.svg"
            />
          </div>
          <div
            className="textcloud"
          >
            <img
              alt="tekst balloon"
              className="textcloud__image"
              src="textcloud.png"
            />
            <div
              className="textcloud__field"
            >
              <p
                className="textcloud__text"
              />
              <svg
                className="continue"
                fill="currentColor"
                height="1em"
                onClick={[Function]}
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 256 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                />
              </svg>
            </div>
          </div>
        </div>
      </section>
    </section>
  </section>,
  ",",
]
`;
